mobile/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,245 @@
+      });
+      snowflake.createStatement({sqlText: `COMMIT;`}).execute();
+
+  } catch(ERROR) {
+      snowflake.createStatement({sqlText: `ROLLBACK;`}).execute();
+
+      // Snowflake error is not very helpful here
+      var err_msg = "Transaction rolled back.Probably failed: " + DML_STATEMENTS + " :Error: ";
+      throw Error(err_msg + ERROR);
+  }
+
+  return "ok. Statements in transaction succeeded."
+
+  $$
+;
+
+
+/*
+  COLUMN_CHECK
+  Checks for mismatched columns between source and target tables.
+  If source table is missing columns, it errors.
+  If AUTOMIGRATE is 'TRUE',
+    it allows the target table to get added the additional columns,if any, of source.
+    Note: Only if extra columns of source are after the common ones, will get added.
+  Since ALTER TABLE is a DDL statement, it will be its own transaction.
+  This means that even if it was placed inside another transaction,
+    it would commit that, and then start another(implicit) for its own execution.
+  So, also it cannot be explicitly rolled back.
+  Input:
+  SOURCE_SCHEMA:  the schema of the source table
+  SOURCE_TABLE:   the source table name
+  TARGET_SCHEMA:  the schema of the target table
+  TARGET_TABLE:   the target table name
+  AUTOMIGRATE:    whether target tabled gets altered if needed (only 'TRUE' enables)
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.column_check(VARCHAR,VARCHAR,VARCHAR,VARCHAR,BOOLEAN);
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.column_check(SOURCE_SCHEMA VARCHAR,
+                                                            SOURCE_TABLE  VARCHAR,
+                                                            TARGET_SCHEMA VARCHAR,
+                                                            TARGET_TABLE  VARCHAR,
+                                                            AUTOMIGRATE   BOOLEAN)
+  RETURNS VARCHAR
+  LANGUAGE JAVASCRIPT
+  AS
+  $$
+
+  var delim = '~';
+  var sourceColumns = list_cols_with_type(SOURCE_SCHEMA,SOURCE_TABLE,delim).split(delim);
+  var targetColumns = list_cols_with_type(TARGET_SCHEMA,TARGET_TABLE,delim).split(delim);
+
+  if (targetColumns.some(notIncludedIn(sourceColumns)) === true) {
+
+      throw "ERROR: Source table is missing column(s) which exist in target table.";
+
+  } else {
+
+      var columnAdditions = sourceColumns.filter(notIncludedIn(targetColumns));
+
+      if (sourceColumns.some(notIncludedIn(targetColumns)) === true) {
+          if (!AUTOMIGRATE) {
+              throw "ERROR: Target table is missing column(s),but automigrate is not enabled.";
+          } else {
+              // enforce order
+              for (var i = 0; i < targetColumns.length; i++) {
+                  if (targetColumns[i] !== sourceColumns[i]) {
+                      throw "ERROR: Can only migrate extra columns of the end of source"
+                  }
+              }
+              add_columns_to(TARGET_SCHEMA, TARGET_TABLE, columnAdditions.join(','));
+              return "ok.Columns added."
```

Comment:
Nit: Being picky here but 'ok' in the success messages (throughout the procedures) seems colloquial and not necessary.